packages/app-project/src/screens/ClassifyPage/components/ClassifierWrapper/ClassifierWrapper.js

```diff
@@ -49,35 +51,36 @@ export default function ClassifierWrapper({
   const ownerSlug = router?.query.owner
   const projectSlug = router?.query.project
 
+  const incrementStats = yourStats?.increment
+  const addRecents = recents?.add
   const onCompleteClassification = useCallback((classification, subject) => {
     const finishedSubject = subject.already_seen || subject.retired
     if (!finishedSubject) {
-      yourStats.increment()
+      incrementStats()
     }
-    recents.add({
+    addRecents({
       favorite: subject.favorite,
       subjectId: subject.id,
       locations: subject.locations
     })
-  }, [recents?.add, yourStats?.increment])
+  }, [addRecents, incrementStats])
 
+  const replaceRoute = router.replace
+  const baseURL = `/${ownerSlug}/${projectSlug}/classify/workflow/${workflowID}`
   const onSubjectChange = useCallback((subject) => {
```

Comment:
I noticed `onSubjectChange` is called every time the locale changes in the url (same behavior exists in #5038 I tested there first).

I don't think this function is intended to run when locale changes, but probably caused because `router` is a dependency of the `useCallback`. This also means the `useEffect` in lib-classifier's ClassifierContainer handling `classifierStore.setOnSubjectChange` runs when locale changes too. Any consequences to this chain of function calls, or is it a harmless/unavoidable side effect?

Reply:
The callback should depend on the `replaceRoute` function, not the whole `router` object.

Reply:
This function should generate a new `onSubjectChange` callback whenever the classifier URL changes. Unfortunately, the callback changes the URL when it runs, so changing the selected subject will generate a new callback for each subject.

Reply:
> I noticed onSubjectChange is called every time the locale changes in the url (same behavior exists in #5038 I tested there first).

I'm not sure I follow this. `onSubjectChange` is the handler for the previous and next buttons in the subject banner. It shouldn't be linked to a props change anywhere.

Reply:
Hmmm. The behavior I noticed is actually logged by the classifier, but I'm trying to understand if it's connected to the `onSubjectChange` function above. For instance, if I visit TESS's classify page, and then change the locale with the dropdown, I see the following logged in the browser console:
```
cleaning up onSubjectChange
setting onSubjectChange
```

The logs are from a `useEffect` in lib-classifier ClassifierContainer with dependencies `[classifierStore.setOnSubjectChange, onSubjectChange]` and the logging happens on every locale change even though the subject remains the same. 

Because `onSubjectChange` is passed as a prop from app-project's ClassifierWrapper, I thought that it must be changing every time locale is updated in app-project. For further testing I added a console log to line 71 inside the `useCallback` of `onSubjectChange` however nothing is logged to the console when locale is changed. 

What would cause the `setting onSubjectChange` logging if the subject is not changing, and only the locale?

Reply:
That’s not running the `onSubjectChange` function. That’s creating the `onSubjectChange` function.

Reply:
The closure’s lexical scope (its dependencies) is changing, which creates a new closure. 
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Closures

Reply:
Incidentally, that was the cause of #4977 too. An old copy of `onSubjectChange` was running, where `subjectID` was still undefined.

Reply:
`onSubjectChange` depends on `router.replace`. Maybe that changes when the locale changes?

This PR is fixing the warning that `router` is missing from the dependency list, so a new callback should be created when the router changes.

Reply:
I’ll have another look at that. It could be that we want `router.replace` to run as a side effect of the subject changing, rather than directly in the callback.

Reply:
I do find the current callback confusing, and hard to debug, because it has so many external dependencies.

Reply:
This example from the `useCallback` docs looks like it’s handling exactly the same problem:
https://react.dev/reference/react/useCallback#preventing-an-effect-from-firing-too-often